cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/autoinstall/HubPackage.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.capability.autoinstall;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Files;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.cdap.cdap.api.artifact.ArtifactRange;
+import io.cdap.cdap.api.artifact.ArtifactVersion;
+import io.cdap.cdap.api.artifact.ArtifactVersionRange;
+import io.cdap.cdap.api.artifact.InvalidArtifactRangeException;
+import io.cdap.cdap.common.ArtifactAlreadyExistsException;
+import io.cdap.cdap.common.id.Id;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import io.cdap.cdap.internal.guava.reflect.TypeToken;
+import io.cdap.cdap.proto.artifact.ArtifactRanges;
+import io.cdap.cdap.proto.id.NamespaceId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static io.cdap.cdap.internal.capability.HttpClient.doGetAsBytes;
+import static io.cdap.cdap.internal.capability.HttpClient.doGetAsString;
+
+/**
+ * Represents a package json in the hub.
+ */
+public class HubPackage {
+  private static final Logger LOG = LoggerFactory.getLogger(HubPackage.class);
+  private static final Gson GSON = new Gson();
+
+  private final String name;
+  private final String version;
+  private final String description;
+  private final String label;
+  private final String author;
+  private final String org;
+  private final String cdapVersion;
+  private final long created;
+  private final boolean beta;
+  private final List<String> categories;
+  private final boolean paid;
+  private final String paidLink;
+  private  transient ArtifactVersionRange versionRange;
```

Comment:
Remove extract space.

Reply:
Why use transient? Can we make these two fields serializable by gson?

Reply:
I kept these as transient because ArtifactVersion, ArtifactVersionRange have additional fields, but the input json just contains a string. Making these fields serializable by gson would require adding custom serde codecs for them which I wanted to avoid. 
WDYT?